controller/src/main/java/io/pravega/controller/store/stream/AbstractStreamMetadataStore.java

```diff
@@ -125,7 +125,8 @@ public OperationContext createContext(String scope, String name) {
                                                    final Executor executor) {
         return getSafeStartingSegmentNumberFor(scope, name)
                 .thenCompose(startingSegmentNumber ->
-                    withCompletion(getStream(scope, name, context).create(configuration, createTimestamp, startingSegmentNumber), executor));
+                    withCompletion(getStream(scope, name, context).create(configuration, createTimestamp, startingSegmentNumber)
```

Comment:
So that I understand, this is adding a new stream twice: once under `scope/` and another under `scope/<set>`?

If so, is there a risk of inconsistencies, like when I'm deleting a stream and I get a concurrent list streams in scope?

Reply:
so delete will "eventually" delete both records. 
So while listing streams, if a delete is happening, its a simple race.. either we retrieve the record under `scope/<set>` before it is deleted or we dont retrieve it. 
Either way we return a list of streams where any of these streams could be deleted as we are delivering these records to the caller. 

There is more nuance in stream creation actually.
we first create stream under `scope` then we add stream to `scope/<set>`. So any partially created stream is not listed by `listStreamsInScope` method. 
Only after streams metadata are created completely does the stream become eligible for listing.
the drawback is, someone could list streams and not find a partially created stream.. attempt to create stream by that name and get a response `EXISTS_CREATING`. 

```
    public enum CreateStatus {
        NEW,
        EXISTS_CREATING,
        EXISTS_ACTIVE,
        FAILED
    }
```